packages/mui-system/src/createStyled.js

```diff
@@ -71,7 +72,31 @@ export default function createStyled(input = {}) {
     slotShouldForwardProp = shouldForwardProp,
     styleFunctionSx = defaultStyleFunctionSx,
   } = input;
+
+  const systemSx = (props) => {
+    const theme = isEmpty(props.theme) ? defaultTheme : props.theme;
+    return styleFunctionSx({ ...props, theme });
+  };
+  systemSx.__mui_systemSx = true;
+
   return (tag, inputOptions = {}) => {
+    /**
```

Comment:
We try to put all emotion/styled-component related logic in the styled-engine packages. Is it possible to move this there, or maybe just export the path to the style rules from the adapter packages? Something like:

```
// this will be set to `__emotion_styles` & `componentStyle.rules` respectivly
import { styleRulesPath } from '@mui/styled-engine';

const styleRules = getValue(tag, styleRulesPath);
if(styleRules) {
// other logic
}
```

This way we won't have anything specific in the system related to emotion/styled-components.

Reply:
sounds good.

Reply:
Done!